core/src/btck/com/view/hud/HealthBar.java

```diff
@@ -0,0 +1,71 @@
+package btck.com.view.hud;
+import btck.com.model.entity.Player;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.Batch;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.GlyphLayout;
+import com.badlogic.gdx.graphics.g2d.freetype.FreeTypeFontGenerator;
+import com.badlogic.gdx.graphics.glutils.ShapeRenderer;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+
+public class HealthBar extends Actor {
+    private ShapeRenderer shapeRenderer;
+    private BitmapFont customFont;
+    private GlyphLayout layout;
+    private Player player;
+
+    public HealthBar(Player player) {
+        this.player = player;
+        this.shapeRenderer = new ShapeRenderer();
+        this.layout = new GlyphLayout();
+        initFont();
+    }
+
+    private void initFont() {
+        FreeTypeFontGenerator generator = new FreeTypeFontGenerator(Gdx.files.internal("HUD/Minecraft.ttf"));
+        FreeTypeFontGenerator.FreeTypeFontParameter params = new FreeTypeFontGenerator.FreeTypeFontParameter();
+        params.size = 24;
+        params.color = Color.WHITE;
+        customFont = generator.generateFont(params);
+        generator.dispose();
+    }
+
+    @Override
+    public void draw(Batch batch, float parentAlpha) {
+        // Dừng batch trước khi vẽ với ShapeRenderer
+        batch.end();
+
+        // Vẽ nền của thanh máu (màu xám đậm)
```

Comment:
thường với các kiểu code mới người ta code theo kiểu hạn chế comment hơn, ít comment

lí do: comment thường không chạy theo kịp với code, đa số các dự án lớn người ta sửa code liên tục, comment k update kịp
ng khác vào đọc thấy cả code + comment -> confuse, giờ tin cái nào, comment thì đang bảo làm ntn, code lại khác
còn trường hợp mà đọc code là đủ hiểu làm gì rồi, lại cần comment nữa -> dupe

solution chung:
đơn giản, các em tách ra 1 hàm riêng để gọi từ hàm to này
vd
drawHealthBar() {

//code

}

-> k cần comment, chỉ có code, đọc từ code ra ý nghĩa